canonical-kubernetes/steps/00_process-providertype/before-deploy

```diff
@@ -6,7 +6,13 @@ set -eux
 
 if [[ "$JUJU_PROVIDERTYPE" == "localhost" ]]; then
     debug "Running pre-deploy for $CONJURE_UP_SPELL"
-    sed "s/##MODEL##/$JUJU_MODEL/" "$(scriptPath)/lxd-profile.yaml" | lxc profile edit "juju-$JUJU_MODEL"
+    lxc_aa_profile="lxc.aa_profile"
+    version=$(lxc --version)
+    version_arr=(${version//./})
```

Comment:
I don't think this is what you want to do, as this removes all `.` from the `${version}`:

```bash
$ version=$(lxc --version)
$ version_arr=(${version//./})
$ echo ${version_arr[0]}
2011
```
As you know, 2011 is greater than 2, so you'll always fall into the condition below.

Reply:
the missing piece here is the space when constructing the `version_arr`:

```diff
- version_arr=(${version//./})
+ version_arr=(${version//./ })
```
Or use IFS and moar wizardry.